ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/SerialTreeParser.java

```diff
@@ -73,8 +76,50 @@ public Node parse(String source) throws TreeParserException {
                 errorMessage = "Code contains syntax error(s).";
             }
         }
-        addDiagnostic(Diagnostic.error(errorMessage));
-        addDiagnostic(Diagnostic.error("Parsing aborted due to errors."));
+        addErrorDiagnostic(errorMessage);
+        addErrorDiagnostic("Parsing aborted due to errors.");
         throw new TreeParserException();
     }
+
+    @Override
+    public Collection<Node> parseDeclarations(String source) throws TreeParserException {
+        try {
+            ModulePartTrial modulePartTrial = new ModulePartTrial(this);
+            ModulePartNode modulePartNode = (ModulePartNode) modulePartTrial.parse(source);
+            List<Node> declarationNodes = new ArrayList<>();
+            modulePartNode.imports().forEach(declarationNodes::add);
+            modulePartNode.members().stream().filter(this::isModuleDeclarationAllowed)
+                    .forEach(declarationNodes::add);
+            return declarationNodes;
+        } catch (ParserTrialFailedException e) {
+            addErrorDiagnostic(e.getMessage());
+            addErrorDiagnostic("Parsing aborted because of errors.");
+            throw new TreeParserException();
+        }
+    }
+
+    /**
+     * Whether the declaration is allowed to be parsed.
+     */
+    private boolean isModuleDeclarationAllowed(ModuleMemberDeclarationNode declarationNode) {
+        if (declarationNode instanceof FunctionDefinitionNode) {
+            String functionName = ((FunctionDefinitionNode) declarationNode).functionName().text();
+            if (functionName.equals("main")) {
+                addWarnDiagnostic("Found 'main' function in the declarations.\n" +
+                        "Discarded 'main' function without loading.");
+                return false;
+            }
+            if (functionName.equals("init")) {
+                addWarnDiagnostic("Found 'init' function in the declarations.\n" +
+                        "Discarded 'init' function without loading.");
```

Comment:
Can we combine these two checks and build the message using the functionName variable?